### ISSUE TITLE
ViroARImageMarker crashes the app

### ISSUE THREAD

**Issue by vasinl124** [viromedia/viro#207]
I followed this to use ARKit 1.5 features
https://docs.viromedia.com/docs/enable-arkit-15-beta-features

Everything seems to work until I add `ViroARImageMarker`  and I got this error
```
ViroARTrackingTargets.createTargets({
  "targetOne" : {
    source : require('./res/targetOne.jpg'),
    orientation : "Up",
    physicalWidth : 0.1 // real world width in meters
  },
});

<ViroARImageMarker target={"targetOne"} >
  <ViroBox position={[0, .25, 0]} scale={[.5, .5, .5] />
</ViroARImageMarker>
```

<img width="868" alt="screen shot 2018-04-01 at 11 07 35 pm" src="https://user-images.githubusercontent.com/6901435/38183954-8305ae16-3601-11e8-935e-cfd4ac409eca.png">

**Comment by mkharibalaji** [comment]
Same Error here as well ?

**Comment by achuvm** [comment]
Hi all,

We tried to reproduce this issue using the variety magazine cover below and it seemed to work for us, so we'll need a bit more information.

1. Are you using an ARKit 1.5 supported device? (iPhone 6S or higher)
2. What is the actual target/asset you are using? Can you attach it here?
3. Can you get more of the error? Xcode -> Window -> Devices and Simulators -> select device -> View Device Logs -> Select the Crash/Error

![variety_magazine](https://user-images.githubusercontent.com/21074927/38206805-a578dfcc-3660-11e8-86df-30a1e3466ba5.jpg)

**Comment by vasinl124** [comment]
1. Are you using an ARKit 1.5 supported device? (iPhone 6S or higher)
I'm using iPhone X with 9.3
2. What is the actual target/asset you are using? Can you attach it here?
I used the image you provided above and still crash the app
3. Can you get more of the error? Xcode -> Window -> Devices and Simulators -> select device -> View Device Logs -> Select the Crash/Error
I can't get the log from it, I don't know why...

**Comment by achuvm** [comment]
Hi @vasinl124,

That device should work.

Can you grab a larger screenshot, ideally with the column in the left showing the stacktrace of the crash? The screenshot you took earlier just shows machine code.

Thanks!

**Comment by vasinl124** [comment]
![screen shot 2018-04-02 at 5 26 17 pm](https://user-images.githubusercontent.com/6901435/38221089-ff195f40-369a-11e8-8c25-230fd2a65fd0.png)

**Comment by achuvm** [comment]
Hi @vasinl124,

Thanks, that's more helpful.

It looks like something in our Renderer is running on an incorrect thread. I'd be curious to see what else is in your Scene that might be causing this issue.

In the meantime, can you see if this scene works for you?
https://gist.github.com/achuvm/669efa3df9093dcfdd03b0736d995b42

The `variety_magazine.jpg` should be placed in a `res/` directory in the same place as the `HelloWorldSceneAR.js`

Thanks,

**Comment by vasinl124** [comment]
Still crashing... with your codes
![screen shot 2018-04-02 at 5 36 06 pm](https://user-images.githubusercontent.com/6901435/38221373-60dd7fb2-369c-11e8-98cb-88b6a1538156.png)

**Comment by achuvm** [comment]
@vasinl124,

That's very strange.

We'll see if we can repro this issue w/ your exact device configuration.

In the meantime, could you try the standard troubleshooting steps:
1. Restarting Xcode/performing a clean on Xcode
2. Restarting your device
3. Creating a new Viro project and copy+pasting the above code I linked above.
4. Try using another iOS device (if possible?)

We'll keep you updated if/when we get more info or are able to reproduce the issue.

Let us know if any of the above suggestions work for you.

Thanks,

**Comment by achuvm** [comment]
Hi @vasinl124,

Is this still an issue for you? We've since updated our platform and no longer require the "beta" steps in order to enable ARKit 1.5 now that it is public.

This also means that you can use our Viro Media App (testbed) to test your application.

Check out the update here: https://docs.viromedia.com/docs/releases

Let us know if you still have the issue.

Thanks,

**Comment by volkova229** [comment]
The same happens to me. Works perfectly before I add `ViroARImageMarker`. The same happens in Viro Media App.

**Comment by volkova229** [comment]
Hello @achuvm 
1. The device in question is iPhone 8 Plus, running iOS 11.3, I checked it multiple times.
2. Yeah, it occurs with the image above, with any image I insert. It also occurs when I install a fresh project and just insert `ViroARImageMarker` into `HelloWorldSceneAR`.
3. 
<img width="1621" alt="trace" src="https://user-images.githubusercontent.com/25868183/38852595-c487ce76-421a-11e8-90e3-09e8df730aac.png">

You're welcome! :)

**Comment by achuvm** [comment]
Hi @volkova229 and @vasinl124,

We've managed to reproduce the issue and should have a fix coming out in our next release (within the week).

It's caused by a slower wifi connection and should only occur on debug builds (race condition between the loading of the target image over wifi and the creation of the component in native).

If you want to work around this for now, you can try performing a release build or, if you have disabled the "AR/VR" selection menu, re-enabling it now in order to "delay" the creation of the `ARImageMarker` target.

Thanks for your help in debugging this issue!

**Comment by vasinl124** [comment]
@achuvm Thanks for the heads up. Good to hear that you found the way to reproduce.

Thanks

**Comment by volkova229** [comment]
@achuvm Thank you very much! :)

**Comment by dam00n** [comment]
This issue has been fixed in our latest release, ViroReact 2.6.0. Release notes and upgrade instructions here -> https://virocore.viromedia.com/docs/3d-objects#section-loading-3d-objects

Closing issue. Please reopen if you are still having problems
